mod-source-record-manager-server/src/main/java/org/folio/services/mappers/processor/functions/NormalizationFunction.java

```diff
@@ -351,6 +353,52 @@ public String apply(RuleExecutionContext context) {
         .map(AlternativeTitleType::getId)
         .orElse(STUB_FIELD_TYPE_ID);
     }
+  },
+
+  SET_ISSUANCE_MODE_ID() {
+
+    @Override
+    public String apply(RuleExecutionContext context) {
+      String subFieldValue = context.getSubFieldValue();
+      char seventhChar = subFieldValue.charAt(6);
+      List<IssuanceMode> issuanceModes = context.getMappingParameters().getIssuanceModes();
+      if (issuanceModes == null || issuanceModes.isEmpty()) {
+        return StringUtils.EMPTY;
+      }
+      String defaultIssuanceModeId = findIssuanceModeId(issuanceModes, IssuanceModeEnum.UNSPECIFIED, StringUtils.EMPTY);
+      return matchIssuanceModeIdViaLeaderSymbol(seventhChar, issuanceModes, defaultIssuanceModeId);
+    }
+
+    private String findIssuanceModeId(List<IssuanceMode> issuanceModes, IssuanceModeEnum issuanceModeType,
+                                      String defaultId) {
+      return issuanceModes.stream()
+        .filter(issuanceMode -> issuanceMode.getName().equals(issuanceModeType.getValue()))
+        .findFirst()
+        .map(IssuanceMode::getId)
+        .orElse(defaultId);
+    }
+
+    private String matchIssuanceModeIdViaLeaderSymbol(char seventhChar, List<IssuanceMode> issuanceModes, String defaultId) {
```

Comment:
move to enum

Reply:
Done. @OleksiiKuzminov could you please review my approach?